\subsection{Sequencing \glsfmtshortpl{crna}}

Sequencing technologies have revolutionized the field of molecular biology,
enabling researchers to explore the complexities of the genome and
transcriptome with unprecedented depth and accuracy.
Traditionally, \gls{rna-seq} has been performed following \gls{polya}
enrichment, a method that selectively isolates \gls{polya} \gls{rna} molecules.
This approach, however, poses significant challenges for the study of
\glspl{crna}, which lack 5' caps and \gls{polya} tails, rendering them
undetectable in conventional sequencing protocols\supercite{guo_expanded_2014}.
As a result, \glspl{crna} have often been overlooked in transcriptomic
analyses, leading to an incomplete understanding of their biological roles and
regulatory mechanisms.

\subsubsection{\Glsfmtlong{trna-seq}}
The advent of \gls{trna-seq} has paved the way for a more comprehensive
investigation of \glspl{crna}.
Unlike \gls{polya} enrichment methods, \gls{trna-seq} captures all \gls{rna}
species, including \glspl{ncrna} and \glspl{crna}, thereby providing a more
holistic view of the transcriptome\supercite{panda_identification_2017}.
This technique has facilitated the identification of numerous \glspl{crna}
across various species and tissues, revealing their high abundance and
stability compared to linear \gls{rna}
counterparts\supercite{liu_circular_2016,cao_expression_2018}.
For instance, studies have shown that \glspl{crna} can be expressed at levels
significantly higher than their linear isomers, highlighting their potential
functional significance in cellular processes\supercite{liu_circular_2016}.
