\section{Sequencing circRNAs}

Sequencing technologies have revolutionized the field of molecular biology,
enabling researchers to explore the complexities of the genome and transcriptome
with unprecedented depth and accuracy. Traditionally, RNA sequencing (RNA-Seq)
has been performed following poly-A enrichment, a method that selectively
isolates polyadenylated mRNA transcripts. This approach, however, poses
significant challenges for the study of circular RNAs (circRNAs), which lack 5'
caps and poly-A tails, rendering them undetectable in conventional sequencing
protocols\supercite{guo_expanded_2014}. As a result, circRNAs have often been overlooked in
transcriptomic analyses, leading to an incomplete understanding of their
biological roles and regulatory mechanisms.

\subsection{Total RNA sequencing (total-RNA-Seq)}
The advent of total-RNA-Seq has paved the way for a more
comprehensive investigation of circRNAs. Unlike poly-A enrichment methods,
total-RNA-Seq captures all RNA species, including non-coding RNAs such as
circRNAs, thereby providing a more holistic view of the
transcriptome\supercite{panda_identification_2017}. This technique has
facilitated the identification of numerous
circRNAs across various species and tissues, revealing their high abundance and
stability compared to linear RNA
counterparts\supercite{liu_circular_2016,cao_expression_2018}. For instance,
studies have shown that circRNAs can be expressed at levels
significantly higher than their linear isomers, highlighting their potential
functional significance in cellular processes\supercite{liu_circular_2016}.
