{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘DESeq2’ was built under R version 4.3.3”\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Warning message:\n",
      "“package ‘S4Vectors’ was built under R version 4.3.3”\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Warning message:\n",
      "“package ‘BiocGenerics’ was built under R version 4.3.2”\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘IRanges’ was built under R version 4.3.3”\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomicRanges’ was built under R version 4.3.3”\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomeInfoDb’ was built under R version 4.3.2”\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Warning message:\n",
      "“package ‘SummarizedExperiment’ was built under R version 4.3.2”\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Warning message:\n",
      "“package ‘MatrixGenerics’ was built under R version 4.3.3”\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Warning message:\n",
      "“package ‘Biobase’ was built under R version 4.3.3”\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘EnhancedVolcano’ was built under R version 4.3.2”\n",
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: ggrepel\n",
      "\n",
      "Warning message:\n",
      "“package ‘clusterProfiler’ was built under R version 4.3.2”\n",
      "\n",
      "\n",
      "clusterProfiler v4.10.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141\n",
      "\n",
      "\n",
      "Attaching package: ‘clusterProfiler’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘enrichplot’ was built under R version 4.3.2”\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Warning message:\n",
      "“package ‘AnnotationDbi’ was built under R version 4.3.2”\n",
      "\n",
      "Attaching package: ‘AnnotationDbi’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘DOSE’ was built under R version 4.3.2”\n",
      "DOSE v3.28.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(DESeq2)\n",
    "library(EnhancedVolcano)\n",
    "library(clusterProfiler)\n",
    "library(enrichplot)\n",
    "library(ggplot2)\n",
    "organism = \"org.Mm.eg.db\"\n",
    "library(organism, character.only = TRUE)\n",
    "library(DOSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "df <- read.table(\"dea/sum.tsv\", header=TRUE, sep=\"\\t\")\n",
    "# Drop duplicates in id column\n",
    "df <- df[!duplicated(df$id),]\n",
    "rownames(df) <- df$id\n",
    "df$id <- NULL\n",
    "\n",
    "annotation <- read.table(\"dea/annotation.bed\", header=FALSE, sep=\"\\t\", col.names = c(\"chr\", \"start\", \"end\", \"name\", \"score\", \"strand\", \"type\", \"gene.id\", \"gene\", \"transcripts\", \"databases\"))\n",
    "# Remove chr, start, end, score, strand, gene.id, transcripts\n",
    "annotation$chr <- NULL\n",
    "annotation$start <- NULL\n",
    "annotation$end <- NULL\n",
    "annotation$score <- NULL\n",
    "annotation$strand <- NULL\n",
    "annotation$gene.id <- NULL\n",
    "annotation$transcripts <- NULL\n",
    "\n",
    "rownames(annotation) <- annotation$name\n",
    "annotation$name <- NULL\n",
    "\n",
    "phenotype <- read.csv(\"dea/phenotype.csv\", header=TRUE, row.names = 1)\n",
    "phenotype$transgene <- as.factor(phenotype$transgene)\n",
    "phenotype$drug <- as.factor(phenotype$drug)\n",
    "\n",
    "# Center and scale age and induction\n",
    "phenotype$age <- scale(phenotype$age)\n",
    "phenotype$induction <- scale(phenotype$induction)\n",
    "\n",
    "df_genes <- read.table(\"dea/gene_tpm.tsv\", sep=\"\\t\", header=TRUE, row.names=1)\n",
    "df_genes$gene_name <- NULL\n",
    "# Order genes columns like phenotype rows\n",
    "df_genes <- df_genes[,rownames(phenotype)]\n",
    "\n",
    "phenotype$esr1 <- scale(as.numeric(df_genes[\"Esr1\",]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "run_analysis <- function(directory, phenotype = NULL, design = NULL, dds = NULL, contrast = NULL, name = NULL) {\n",
    "    # Fail if both contrast and name are NULL\n",
    "    if (is.null(contrast) && is.null(name)) {\n",
    "        stop(\"Both contrast and name cannot be NULL\")\n",
    "    }\n",
    "    # Fail if none of contrast and name are NULL\n",
    "    if (!is.null(contrast) && !is.null(name)) {\n",
    "        stop(\"Both contrast and name cannot be provided\")\n",
    "    }\n",
    "\n",
    "    if (is.null(phenotype) && is.null(design) && is.null(dds)) {\n",
    "        stop(\"Either phenotype and design or dds must be provided\")\n",
    "    }\n",
    "\n",
    "    if (is.null(phenotype) != is.null(design)) {\n",
    "        stop(\"Both phenotype and design must be provided\")\n",
    "    }\n",
    "\n",
    "    if (!is.null(dds) && !is.null(phenotype)) {\n",
    "        stop(\"Either dds or phenotype and design must be provided\")\n",
    "    }\n",
    "\n",
    "    alpha <- 0.05\n",
    "\n",
    "    if (is.null(dds)) {\n",
    "        dds <- DESeqDataSetFromMatrix(countData = df[, rownames(phenotype)],\n",
    "                                    colData = phenotype,\n",
    "                                    design = design)\n",
    "        dds <- DESeq(dds)\n",
    "    }\n",
    "\n",
    "    if (!is.null(contrast)) {\n",
    "        res <- results(dds, contrast = contrast, alpha = alpha, lfcThreshold = 2, altHypothesis = \"greaterAbs\")\n",
    "    } else {\n",
    "        res <- results(dds, name = name, alpha = alpha, lfcThreshold = 2, altHypothesis = \"greaterAbs\")\n",
    "    }\n",
    "    res <- res[order(res$padj),]\n",
    "    res <- cbind(res, annotation[rownames(res),])\n",
    "\n",
    "    # Recreate directory if it exists\n",
    "    if (dir.exists(directory)) {\n",
    "        unlink(directory, recursive = TRUE)\n",
    "    }\n",
    "    dir.create(directory, showWarnings = FALSE)\n",
    "\n",
    "    write.table(res, file=paste0(directory, \"/res.tsv\"), sep=\"\\t\", col.names=NA, row.names=TRUE, quote=FALSE)\n",
    "\n",
    "    EnhancedVolcano(res,\n",
    "        x='log2FoldChange',\n",
    "        y='padj',\n",
    "        lab=res$gene,\n",
    "        pCutoff = alpha)\n",
    "    ggsave(paste0(directory, \"/volcano.png\"))\n",
    "\n",
    "    # Keep only rows without any NA\n",
    "    res <- res[complete.cases(res),]\n",
    "\n",
    "    sign <- res[res$padj < alpha,]\n",
    "\n",
    "    genes <- unique(unlist(strsplit(sign$gene, \",\")))\n",
    "    write.table(genes, file=paste0(directory, \"/genes.txt\"), row.names=FALSE, col.names=FALSE, quote=FALSE)\n",
    "    db <- unique(unlist(strsplit(sign$databases, \",\")))\n",
    "    write.table(db, file=paste0(directory, \"/db.txt\"), row.names=FALSE, col.names=FALSE, quote=FALSE)\n",
    "\n",
    "    gene_list <- sign$log2FoldChange\n",
    "    names(gene_list) <- sign$gene\n",
    "    gene_list = sort(gene_list, decreasing = TRUE)\n",
    "\n",
    "    gse <- gseGO(geneList=gene_list,\n",
    "            ont =\"ALL\", \n",
    "            keyType = \"SYMBOL\",\n",
    "            pvalueCutoff = 0.05,\n",
    "            verbose = TRUE,\n",
    "            OrgDb = organism,\n",
    "            pAdjustMethod = \"BH\")\n",
    "\n",
    "    # If no significant GO terms, return\n",
    "    if (nrow(as.data.frame(gse)) > 0) {\n",
    "        dotplot(gse, showCategory=10, split=\".sign\") + facet_grid(.~.sign)\n",
    "        ggsave(paste0(directory, \"/dot.png\"))\n",
    "\n",
    "        gse <- pairwise_termsim(gse)\n",
    "        emapplot(gse, showCategory = 10)\n",
    "        ggsave(paste0(directory, \"/emap.png\"))\n",
    "\n",
    "        cnetplot(gse, categorySize=\"pvalue\", foldChange=gene_list, showCategory = 3)\n",
    "        ggsave(paste0(directory, \"/cnet.png\"))\n",
    "\n",
    "        ridgeplot(gse) + labs(x = \"enrichment distribution\")\n",
    "        ggsave(paste0(directory, \"/ridge.png\"))\n",
    "    } else {\n",
    "        print(\"No significant GO terms\")\n",
    "    }\n",
    "\n",
    "    return(dds)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Aging"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "15 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (83.56% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Warning message in cnetplot.enrichResult(x, ...):\n",
      "“Use 'color.params = list(foldChange = your_value)' instead of 'foldChange'.\n",
      " The foldChange parameter will be removed in the next version.”\n",
      "\u001b[1m\u001b[22mScale for \u001b[32msize\u001b[39m is already present.\n",
      "Adding another scale for \u001b[32msize\u001b[39m, which will replace the existing scale.\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Warning message:\n",
      "“ggrepel: 9 unlabeled data points (too many overlaps). Consider increasing max.overlaps”\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Picking joint bandwidth of 4.49\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dds_esr1 <- run_analysis(\n",
    "    \"dea/age\",\n",
    "    phenotype = phenotype,\n",
    "    design = ~ age + transgene + induction + drug + esr1,\n",
    "    name = \"age\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ESR1 association"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 56620 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>correlation</th><th scope=col>p.value</th><th scope=col>p.adj</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>chr10:117537469-117541134:+</th><td>-0.3825553</td><td>0.0009118705</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr10:117537470-117541134:-</th><td>-0.3825553</td><td>0.0009118705</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr3:108030857-108031153:-</th><td>-0.3763173</td><td>0.0011223173</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr3:108030858-108031153:+</th><td>-0.3763173</td><td>0.0011223173</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr12:3501661-3507209:+</th><td>-0.3728414</td><td>0.0012577724</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr12:3501662-3507209:-</th><td>-0.3728414</td><td>0.0012577724</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr12:13519842-13534045:+</th><td>-0.3603073</td><td>0.0018776093</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr12:13519842-13534045:-</th><td>-0.3603073</td><td>0.0018776093</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr6:17844907-17855027:-</th><td>-0.3582871</td><td>0.0019998994</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr6:17844908-17855027:+</th><td>-0.3582871</td><td>0.0019998994</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chrX:156333927-156344497:+</th><td>-0.3538968</td><td>0.0022906418</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chrX:156333928-156344497:-</th><td>-0.3538968</td><td>0.0022906418</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr18:73790265-73795938:-</th><td>-0.3521414</td><td>0.0024171096</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr18:73790266-73795938:+</th><td>-0.3521414</td><td>0.0024171096</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr1:34482943-34483139:-</th><td>-0.3411083</td><td>0.0033653357</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr19:60759453-60766585:+</th><td>-0.3374783</td><td>0.0037429571</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr19:60759453-60766585:-</th><td>-0.3374783</td><td>0.0037429571</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr14:33124580-33132981:-</th><td>-0.3373424</td><td>0.0037578017</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr14:33124581-33132981:+</th><td>-0.3373424</td><td>0.0037578017</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr9:72499032-72517670:+</th><td>-0.3366938</td><td>0.0038293600</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr9:72499032-72517670:-</th><td>-0.3366938</td><td>0.0038293600</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr19:57638689-57645670:+</th><td>-0.3348893</td><td>0.0040348467</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr19:57638689-57645670:-</th><td>-0.3348893</td><td>0.0040348467</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr12:91331133-91336970:+</th><td>-0.3307806</td><td>0.0045396710</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr12:91331134-91336970:-</th><td>-0.3307806</td><td>0.0045396710</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr10:98822752-98832844:-</th><td>-0.3305739</td><td>0.0045664825</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr10:98822753-98832844:+</th><td>-0.3305739</td><td>0.0045664825</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr10:121761720-121767414:+</th><td>-0.3298647</td><td>0.0046595496</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr10:121761720-121767414:-</th><td>-0.3298647</td><td>0.0046595496</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>chr14:79171105-79263517:+</th><td>-0.3281036</td><td>0.0048979776</td><td>0.9200633</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>chr19:56368638-56368745:+</th><td>0.3624252</td><td>0.0017566563</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr10:26721927-26736728:+</th><td>0.3629644</td><td>0.0017270009</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr10:26721928-26736728:-</th><td>0.3629644</td><td>0.0017270009</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr2:37422528-37433243:+</th><td>0.3640561</td><td>0.0016683395</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr2:37422528-37433243:-</th><td>0.3640561</td><td>0.0016683395</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr12:85354703-85358220:-</th><td>0.3678362</td><td>0.0014788215</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr12:85354704-85358220:+</th><td>0.3678362</td><td>0.0014788215</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr4:134367597-134367694:-</th><td>0.3678362</td><td>0.0014788215</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr4:134367598-134367694:+</th><td>0.3678362</td><td>0.0014788215</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr15:11234027-11241631:+</th><td>0.3699613</td><td>0.0013810111</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr15:11234027-11241631:-</th><td>0.3699613</td><td>0.0013810111</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr18:56678036-56681651:+</th><td>0.3708500</td><td>0.0013418701</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr18:56678037-56681651:-</th><td>0.3708500</td><td>0.0013418701</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr1:93347790-93369972:-</th><td>0.3709959</td><td>0.0013355415</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr1:93347791-93369972:+</th><td>0.3709959</td><td>0.0013355415</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr4:86743154-86744317:-</th><td>0.3722684</td><td>0.0012814682</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr4:86743155-86744317:+</th><td>0.3722684</td><td>0.0012814682</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr11:86551948-86556704:+</th><td>0.3727350</td><td>0.0012621435</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr11:86551948-86556704:-</th><td>0.3727350</td><td>0.0012621435</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr4:150490653-150516050:+</th><td>0.3748595</td><td>0.0011774320</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr4:150490653-150516050:-</th><td>0.3748595</td><td>0.0011774320</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr5:87934481-87936417:-</th><td>0.3807488</td><td>0.0009687936</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr14:14712667-14730314:-</th><td>0.4157796</td><td>0.0002810410</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr14:14712668-14730314:+</th><td>0.4157796</td><td>0.0002810410</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr14:26313388-26340559:+</th><td>0.4157796</td><td>0.0002810410</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr14:26313388-26340559:-</th><td>0.4157796</td><td>0.0002810410</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr19:28508272-28517808:-</th><td>0.4161321</td><td>0.0002773681</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr19:28508273-28517808:+</th><td>0.4161321</td><td>0.0002773681</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr16:15443425-15444128:+</th><td>0.4233722</td><td>0.0002110251</td><td>0.8156611</td></tr>\n",
       "\t<tr><th scope=row>chr16:15443426-15444128:-</th><td>0.4233722</td><td>0.0002110251</td><td>0.8156611</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 56620 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & correlation & p.value & p.adj\\\\\n",
       "  & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tchr10:117537469-117541134:+ & -0.3825553 & 0.0009118705 & 0.8156611\\\\\n",
       "\tchr10:117537470-117541134:- & -0.3825553 & 0.0009118705 & 0.8156611\\\\\n",
       "\tchr3:108030857-108031153:- & -0.3763173 & 0.0011223173 & 0.8156611\\\\\n",
       "\tchr3:108030858-108031153:+ & -0.3763173 & 0.0011223173 & 0.8156611\\\\\n",
       "\tchr12:3501661-3507209:+ & -0.3728414 & 0.0012577724 & 0.8156611\\\\\n",
       "\tchr12:3501662-3507209:- & -0.3728414 & 0.0012577724 & 0.8156611\\\\\n",
       "\tchr12:13519842-13534045:+ & -0.3603073 & 0.0018776093 & 0.8156611\\\\\n",
       "\tchr12:13519842-13534045:- & -0.3603073 & 0.0018776093 & 0.8156611\\\\\n",
       "\tchr6:17844907-17855027:- & -0.3582871 & 0.0019998994 & 0.8156611\\\\\n",
       "\tchr6:17844908-17855027:+ & -0.3582871 & 0.0019998994 & 0.8156611\\\\\n",
       "\tchrX:156333927-156344497:+ & -0.3538968 & 0.0022906418 & 0.8156611\\\\\n",
       "\tchrX:156333928-156344497:- & -0.3538968 & 0.0022906418 & 0.8156611\\\\\n",
       "\tchr18:73790265-73795938:- & -0.3521414 & 0.0024171096 & 0.8156611\\\\\n",
       "\tchr18:73790266-73795938:+ & -0.3521414 & 0.0024171096 & 0.8156611\\\\\n",
       "\tchr1:34482943-34483139:- & -0.3411083 & 0.0033653357 & 0.9200633\\\\\n",
       "\tchr19:60759453-60766585:+ & -0.3374783 & 0.0037429571 & 0.9200633\\\\\n",
       "\tchr19:60759453-60766585:- & -0.3374783 & 0.0037429571 & 0.9200633\\\\\n",
       "\tchr14:33124580-33132981:- & -0.3373424 & 0.0037578017 & 0.9200633\\\\\n",
       "\tchr14:33124581-33132981:+ & -0.3373424 & 0.0037578017 & 0.9200633\\\\\n",
       "\tchr9:72499032-72517670:+ & -0.3366938 & 0.0038293600 & 0.9200633\\\\\n",
       "\tchr9:72499032-72517670:- & -0.3366938 & 0.0038293600 & 0.9200633\\\\\n",
       "\tchr19:57638689-57645670:+ & -0.3348893 & 0.0040348467 & 0.9200633\\\\\n",
       "\tchr19:57638689-57645670:- & -0.3348893 & 0.0040348467 & 0.9200633\\\\\n",
       "\tchr12:91331133-91336970:+ & -0.3307806 & 0.0045396710 & 0.9200633\\\\\n",
       "\tchr12:91331134-91336970:- & -0.3307806 & 0.0045396710 & 0.9200633\\\\\n",
       "\tchr10:98822752-98832844:- & -0.3305739 & 0.0045664825 & 0.9200633\\\\\n",
       "\tchr10:98822753-98832844:+ & -0.3305739 & 0.0045664825 & 0.9200633\\\\\n",
       "\tchr10:121761720-121767414:+ & -0.3298647 & 0.0046595496 & 0.9200633\\\\\n",
       "\tchr10:121761720-121767414:- & -0.3298647 & 0.0046595496 & 0.9200633\\\\\n",
       "\tchr14:79171105-79263517:+ & -0.3281036 & 0.0048979776 & 0.9200633\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tchr19:56368638-56368745:+ & 0.3624252 & 0.0017566563 & 0.8156611\\\\\n",
       "\tchr10:26721927-26736728:+ & 0.3629644 & 0.0017270009 & 0.8156611\\\\\n",
       "\tchr10:26721928-26736728:- & 0.3629644 & 0.0017270009 & 0.8156611\\\\\n",
       "\tchr2:37422528-37433243:+ & 0.3640561 & 0.0016683395 & 0.8156611\\\\\n",
       "\tchr2:37422528-37433243:- & 0.3640561 & 0.0016683395 & 0.8156611\\\\\n",
       "\tchr12:85354703-85358220:- & 0.3678362 & 0.0014788215 & 0.8156611\\\\\n",
       "\tchr12:85354704-85358220:+ & 0.3678362 & 0.0014788215 & 0.8156611\\\\\n",
       "\tchr4:134367597-134367694:- & 0.3678362 & 0.0014788215 & 0.8156611\\\\\n",
       "\tchr4:134367598-134367694:+ & 0.3678362 & 0.0014788215 & 0.8156611\\\\\n",
       "\tchr15:11234027-11241631:+ & 0.3699613 & 0.0013810111 & 0.8156611\\\\\n",
       "\tchr15:11234027-11241631:- & 0.3699613 & 0.0013810111 & 0.8156611\\\\\n",
       "\tchr18:56678036-56681651:+ & 0.3708500 & 0.0013418701 & 0.8156611\\\\\n",
       "\tchr18:56678037-56681651:- & 0.3708500 & 0.0013418701 & 0.8156611\\\\\n",
       "\tchr1:93347790-93369972:- & 0.3709959 & 0.0013355415 & 0.8156611\\\\\n",
       "\tchr1:93347791-93369972:+ & 0.3709959 & 0.0013355415 & 0.8156611\\\\\n",
       "\tchr4:86743154-86744317:- & 0.3722684 & 0.0012814682 & 0.8156611\\\\\n",
       "\tchr4:86743155-86744317:+ & 0.3722684 & 0.0012814682 & 0.8156611\\\\\n",
       "\tchr11:86551948-86556704:+ & 0.3727350 & 0.0012621435 & 0.8156611\\\\\n",
       "\tchr11:86551948-86556704:- & 0.3727350 & 0.0012621435 & 0.8156611\\\\\n",
       "\tchr4:150490653-150516050:+ & 0.3748595 & 0.0011774320 & 0.8156611\\\\\n",
       "\tchr4:150490653-150516050:- & 0.3748595 & 0.0011774320 & 0.8156611\\\\\n",
       "\tchr5:87934481-87936417:- & 0.3807488 & 0.0009687936 & 0.8156611\\\\\n",
       "\tchr14:14712667-14730314:- & 0.4157796 & 0.0002810410 & 0.8156611\\\\\n",
       "\tchr14:14712668-14730314:+ & 0.4157796 & 0.0002810410 & 0.8156611\\\\\n",
       "\tchr14:26313388-26340559:+ & 0.4157796 & 0.0002810410 & 0.8156611\\\\\n",
       "\tchr14:26313388-26340559:- & 0.4157796 & 0.0002810410 & 0.8156611\\\\\n",
       "\tchr19:28508272-28517808:- & 0.4161321 & 0.0002773681 & 0.8156611\\\\\n",
       "\tchr19:28508273-28517808:+ & 0.4161321 & 0.0002773681 & 0.8156611\\\\\n",
       "\tchr16:15443425-15444128:+ & 0.4233722 & 0.0002110251 & 0.8156611\\\\\n",
       "\tchr16:15443426-15444128:- & 0.4233722 & 0.0002110251 & 0.8156611\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 56620 × 3\n",
       "\n",
       "| <!--/--> | correlation &lt;dbl&gt; | p.value &lt;dbl&gt; | p.adj &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| chr10:117537469-117541134:+ | -0.3825553 | 0.0009118705 | 0.8156611 |\n",
       "| chr10:117537470-117541134:- | -0.3825553 | 0.0009118705 | 0.8156611 |\n",
       "| chr3:108030857-108031153:- | -0.3763173 | 0.0011223173 | 0.8156611 |\n",
       "| chr3:108030858-108031153:+ | -0.3763173 | 0.0011223173 | 0.8156611 |\n",
       "| chr12:3501661-3507209:+ | -0.3728414 | 0.0012577724 | 0.8156611 |\n",
       "| chr12:3501662-3507209:- | -0.3728414 | 0.0012577724 | 0.8156611 |\n",
       "| chr12:13519842-13534045:+ | -0.3603073 | 0.0018776093 | 0.8156611 |\n",
       "| chr12:13519842-13534045:- | -0.3603073 | 0.0018776093 | 0.8156611 |\n",
       "| chr6:17844907-17855027:- | -0.3582871 | 0.0019998994 | 0.8156611 |\n",
       "| chr6:17844908-17855027:+ | -0.3582871 | 0.0019998994 | 0.8156611 |\n",
       "| chrX:156333927-156344497:+ | -0.3538968 | 0.0022906418 | 0.8156611 |\n",
       "| chrX:156333928-156344497:- | -0.3538968 | 0.0022906418 | 0.8156611 |\n",
       "| chr18:73790265-73795938:- | -0.3521414 | 0.0024171096 | 0.8156611 |\n",
       "| chr18:73790266-73795938:+ | -0.3521414 | 0.0024171096 | 0.8156611 |\n",
       "| chr1:34482943-34483139:- | -0.3411083 | 0.0033653357 | 0.9200633 |\n",
       "| chr19:60759453-60766585:+ | -0.3374783 | 0.0037429571 | 0.9200633 |\n",
       "| chr19:60759453-60766585:- | -0.3374783 | 0.0037429571 | 0.9200633 |\n",
       "| chr14:33124580-33132981:- | -0.3373424 | 0.0037578017 | 0.9200633 |\n",
       "| chr14:33124581-33132981:+ | -0.3373424 | 0.0037578017 | 0.9200633 |\n",
       "| chr9:72499032-72517670:+ | -0.3366938 | 0.0038293600 | 0.9200633 |\n",
       "| chr9:72499032-72517670:- | -0.3366938 | 0.0038293600 | 0.9200633 |\n",
       "| chr19:57638689-57645670:+ | -0.3348893 | 0.0040348467 | 0.9200633 |\n",
       "| chr19:57638689-57645670:- | -0.3348893 | 0.0040348467 | 0.9200633 |\n",
       "| chr12:91331133-91336970:+ | -0.3307806 | 0.0045396710 | 0.9200633 |\n",
       "| chr12:91331134-91336970:- | -0.3307806 | 0.0045396710 | 0.9200633 |\n",
       "| chr10:98822752-98832844:- | -0.3305739 | 0.0045664825 | 0.9200633 |\n",
       "| chr10:98822753-98832844:+ | -0.3305739 | 0.0045664825 | 0.9200633 |\n",
       "| chr10:121761720-121767414:+ | -0.3298647 | 0.0046595496 | 0.9200633 |\n",
       "| chr10:121761720-121767414:- | -0.3298647 | 0.0046595496 | 0.9200633 |\n",
       "| chr14:79171105-79263517:+ | -0.3281036 | 0.0048979776 | 0.9200633 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| chr19:56368638-56368745:+ | 0.3624252 | 0.0017566563 | 0.8156611 |\n",
       "| chr10:26721927-26736728:+ | 0.3629644 | 0.0017270009 | 0.8156611 |\n",
       "| chr10:26721928-26736728:- | 0.3629644 | 0.0017270009 | 0.8156611 |\n",
       "| chr2:37422528-37433243:+ | 0.3640561 | 0.0016683395 | 0.8156611 |\n",
       "| chr2:37422528-37433243:- | 0.3640561 | 0.0016683395 | 0.8156611 |\n",
       "| chr12:85354703-85358220:- | 0.3678362 | 0.0014788215 | 0.8156611 |\n",
       "| chr12:85354704-85358220:+ | 0.3678362 | 0.0014788215 | 0.8156611 |\n",
       "| chr4:134367597-134367694:- | 0.3678362 | 0.0014788215 | 0.8156611 |\n",
       "| chr4:134367598-134367694:+ | 0.3678362 | 0.0014788215 | 0.8156611 |\n",
       "| chr15:11234027-11241631:+ | 0.3699613 | 0.0013810111 | 0.8156611 |\n",
       "| chr15:11234027-11241631:- | 0.3699613 | 0.0013810111 | 0.8156611 |\n",
       "| chr18:56678036-56681651:+ | 0.3708500 | 0.0013418701 | 0.8156611 |\n",
       "| chr18:56678037-56681651:- | 0.3708500 | 0.0013418701 | 0.8156611 |\n",
       "| chr1:93347790-93369972:- | 0.3709959 | 0.0013355415 | 0.8156611 |\n",
       "| chr1:93347791-93369972:+ | 0.3709959 | 0.0013355415 | 0.8156611 |\n",
       "| chr4:86743154-86744317:- | 0.3722684 | 0.0012814682 | 0.8156611 |\n",
       "| chr4:86743155-86744317:+ | 0.3722684 | 0.0012814682 | 0.8156611 |\n",
       "| chr11:86551948-86556704:+ | 0.3727350 | 0.0012621435 | 0.8156611 |\n",
       "| chr11:86551948-86556704:- | 0.3727350 | 0.0012621435 | 0.8156611 |\n",
       "| chr4:150490653-150516050:+ | 0.3748595 | 0.0011774320 | 0.8156611 |\n",
       "| chr4:150490653-150516050:- | 0.3748595 | 0.0011774320 | 0.8156611 |\n",
       "| chr5:87934481-87936417:- | 0.3807488 | 0.0009687936 | 0.8156611 |\n",
       "| chr14:14712667-14730314:- | 0.4157796 | 0.0002810410 | 0.8156611 |\n",
       "| chr14:14712668-14730314:+ | 0.4157796 | 0.0002810410 | 0.8156611 |\n",
       "| chr14:26313388-26340559:+ | 0.4157796 | 0.0002810410 | 0.8156611 |\n",
       "| chr14:26313388-26340559:- | 0.4157796 | 0.0002810410 | 0.8156611 |\n",
       "| chr19:28508272-28517808:- | 0.4161321 | 0.0002773681 | 0.8156611 |\n",
       "| chr19:28508273-28517808:+ | 0.4161321 | 0.0002773681 | 0.8156611 |\n",
       "| chr16:15443425-15444128:+ | 0.4233722 | 0.0002110251 | 0.8156611 |\n",
       "| chr16:15443426-15444128:- | 0.4233722 | 0.0002110251 | 0.8156611 |\n",
       "\n"
      ],
      "text/plain": [
       "                            correlation p.value      p.adj    \n",
       "chr10:117537469-117541134:+ -0.3825553  0.0009118705 0.8156611\n",
       "chr10:117537470-117541134:- -0.3825553  0.0009118705 0.8156611\n",
       "chr3:108030857-108031153:-  -0.3763173  0.0011223173 0.8156611\n",
       "chr3:108030858-108031153:+  -0.3763173  0.0011223173 0.8156611\n",
       "chr12:3501661-3507209:+     -0.3728414  0.0012577724 0.8156611\n",
       "chr12:3501662-3507209:-     -0.3728414  0.0012577724 0.8156611\n",
       "chr12:13519842-13534045:+   -0.3603073  0.0018776093 0.8156611\n",
       "chr12:13519842-13534045:-   -0.3603073  0.0018776093 0.8156611\n",
       "chr6:17844907-17855027:-    -0.3582871  0.0019998994 0.8156611\n",
       "chr6:17844908-17855027:+    -0.3582871  0.0019998994 0.8156611\n",
       "chrX:156333927-156344497:+  -0.3538968  0.0022906418 0.8156611\n",
       "chrX:156333928-156344497:-  -0.3538968  0.0022906418 0.8156611\n",
       "chr18:73790265-73795938:-   -0.3521414  0.0024171096 0.8156611\n",
       "chr18:73790266-73795938:+   -0.3521414  0.0024171096 0.8156611\n",
       "chr1:34482943-34483139:-    -0.3411083  0.0033653357 0.9200633\n",
       "chr19:60759453-60766585:+   -0.3374783  0.0037429571 0.9200633\n",
       "chr19:60759453-60766585:-   -0.3374783  0.0037429571 0.9200633\n",
       "chr14:33124580-33132981:-   -0.3373424  0.0037578017 0.9200633\n",
       "chr14:33124581-33132981:+   -0.3373424  0.0037578017 0.9200633\n",
       "chr9:72499032-72517670:+    -0.3366938  0.0038293600 0.9200633\n",
       "chr9:72499032-72517670:-    -0.3366938  0.0038293600 0.9200633\n",
       "chr19:57638689-57645670:+   -0.3348893  0.0040348467 0.9200633\n",
       "chr19:57638689-57645670:-   -0.3348893  0.0040348467 0.9200633\n",
       "chr12:91331133-91336970:+   -0.3307806  0.0045396710 0.9200633\n",
       "chr12:91331134-91336970:-   -0.3307806  0.0045396710 0.9200633\n",
       "chr10:98822752-98832844:-   -0.3305739  0.0045664825 0.9200633\n",
       "chr10:98822753-98832844:+   -0.3305739  0.0045664825 0.9200633\n",
       "chr10:121761720-121767414:+ -0.3298647  0.0046595496 0.9200633\n",
       "chr10:121761720-121767414:- -0.3298647  0.0046595496 0.9200633\n",
       "chr14:79171105-79263517:+   -0.3281036  0.0048979776 0.9200633\n",
       "⋮                           ⋮           ⋮            ⋮        \n",
       "chr19:56368638-56368745:+   0.3624252   0.0017566563 0.8156611\n",
       "chr10:26721927-26736728:+   0.3629644   0.0017270009 0.8156611\n",
       "chr10:26721928-26736728:-   0.3629644   0.0017270009 0.8156611\n",
       "chr2:37422528-37433243:+    0.3640561   0.0016683395 0.8156611\n",
       "chr2:37422528-37433243:-    0.3640561   0.0016683395 0.8156611\n",
       "chr12:85354703-85358220:-   0.3678362   0.0014788215 0.8156611\n",
       "chr12:85354704-85358220:+   0.3678362   0.0014788215 0.8156611\n",
       "chr4:134367597-134367694:-  0.3678362   0.0014788215 0.8156611\n",
       "chr4:134367598-134367694:+  0.3678362   0.0014788215 0.8156611\n",
       "chr15:11234027-11241631:+   0.3699613   0.0013810111 0.8156611\n",
       "chr15:11234027-11241631:-   0.3699613   0.0013810111 0.8156611\n",
       "chr18:56678036-56681651:+   0.3708500   0.0013418701 0.8156611\n",
       "chr18:56678037-56681651:-   0.3708500   0.0013418701 0.8156611\n",
       "chr1:93347790-93369972:-    0.3709959   0.0013355415 0.8156611\n",
       "chr1:93347791-93369972:+    0.3709959   0.0013355415 0.8156611\n",
       "chr4:86743154-86744317:-    0.3722684   0.0012814682 0.8156611\n",
       "chr4:86743155-86744317:+    0.3722684   0.0012814682 0.8156611\n",
       "chr11:86551948-86556704:+   0.3727350   0.0012621435 0.8156611\n",
       "chr11:86551948-86556704:-   0.3727350   0.0012621435 0.8156611\n",
       "chr4:150490653-150516050:+  0.3748595   0.0011774320 0.8156611\n",
       "chr4:150490653-150516050:-  0.3748595   0.0011774320 0.8156611\n",
       "chr5:87934481-87936417:-    0.3807488   0.0009687936 0.8156611\n",
       "chr14:14712667-14730314:-   0.4157796   0.0002810410 0.8156611\n",
       "chr14:14712668-14730314:+   0.4157796   0.0002810410 0.8156611\n",
       "chr14:26313388-26340559:+   0.4157796   0.0002810410 0.8156611\n",
       "chr14:26313388-26340559:-   0.4157796   0.0002810410 0.8156611\n",
       "chr19:28508272-28517808:-   0.4161321   0.0002773681 0.8156611\n",
       "chr19:28508273-28517808:+   0.4161321   0.0002773681 0.8156611\n",
       "chr16:15443425-15444128:+   0.4233722   0.0002110251 0.8156611\n",
       "chr16:15443426-15444128:-   0.4233722   0.0002110251 0.8156611"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "correlations <- t(apply(df, 1, function(row) {\n",
    "    res <- cor.test(as.numeric(row), phenotype$esr1)\n",
    "    return(c(res$estimate, res$p.value))\n",
    "}))\n",
    "correlations <- as.data.frame(correlations)\n",
    "colnames(correlations) <- c(\"correlation\", \"p.value\")\n",
    "correlations$p.adj <- p.adjust(correlations$p.value, method = \"BH\")\n",
    "correlations <- correlations[order(correlations$correlation),]\n",
    "correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (87.45% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Warning message in cnetplot.enrichResult(x, ...):\n",
      "“Use 'color.params = list(foldChange = your_value)' instead of 'foldChange'.\n",
      " The foldChange parameter will be removed in the next version.”\n",
      "\u001b[1m\u001b[22mScale for \u001b[32msize\u001b[39m is already present.\n",
      "Adding another scale for \u001b[32msize\u001b[39m, which will replace the existing scale.\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Warning message:\n",
      "“ggrepel: 770 unlabeled data points (too many overlaps). Consider increasing max.overlaps”\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Picking joint bandwidth of 1.19\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 56620 72 \n",
       "metadata(1): version\n",
       "assays(4): counts mu H cooks\n",
       "rownames(56620): chr1:3729265-3729444:+ chr1:4212834-4337843:- ...\n",
       "  chrX_GL456233v2_random:409325-409604:-\n",
       "  chrX_GL456233v2_random:409325-422124:-\n",
       "rowData names(42): baseMean baseVar ... deviance maxCooks\n",
       "colnames(72): aging_12m_ESR1_no_1 aging_12m_ESR1_no_2 ...\n",
       "  antiHormonal_18m_ESR1_no_2 antiHormonal_18m_ESR1_no_3\n",
       "colData names(8): condition age ... esr1 sizeFactor"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "run_analysis(\n",
    "    \"dea/esr1\",\n",
    "    dds = dds_esr1,\n",
    "    name = \"esr1\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Drug effects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Tamoxifen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (52.48% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"No significant GO terms\"\n"
     ]
    }
   ],
   "source": [
    "dds_no_esr1 <- run_analysis(\n",
    "    \"dea/tamoxifen\",\n",
    "    phenotype = phenotype,\n",
    "    design = ~ age + transgene + induction + drug,\n",
    "    contrast=c(\"drug\", \"tamoxifen\", \"no\")\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Letrozole"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (64.39% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"No significant GO terms\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 56620 72 \n",
       "metadata(1): version\n",
       "assays(4): counts mu H cooks\n",
       "rownames(56620): chr1:3729265-3729444:+ chr1:4212834-4337843:- ...\n",
       "  chrX_GL456233v2_random:409325-409604:-\n",
       "  chrX_GL456233v2_random:409325-422124:-\n",
       "rowData names(38): baseMean baseVar ... deviance maxCooks\n",
       "colnames(72): aging_12m_ESR1_no_1 aging_12m_ESR1_no_2 ...\n",
       "  antiHormonal_18m_ESR1_no_2 antiHormonal_18m_ESR1_no_3\n",
       "colData names(8): condition age ... esr1 sizeFactor"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "run_analysis(\n",
    "    \"dea/letrozole\",\n",
    "    dds = dds_no_esr1,\n",
    "    contrast=c(\"drug\", \"letrozole\", \"no\")\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
