{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘DESeq2’ was built under R version 4.3.3”\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Warning message:\n",
      "“package ‘S4Vectors’ was built under R version 4.3.3”\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Warning message:\n",
      "“package ‘BiocGenerics’ was built under R version 4.3.2”\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘IRanges’ was built under R version 4.3.3”\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomicRanges’ was built under R version 4.3.3”\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomeInfoDb’ was built under R version 4.3.2”\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Warning message:\n",
      "“package ‘SummarizedExperiment’ was built under R version 4.3.2”\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Warning message:\n",
      "“package ‘MatrixGenerics’ was built under R version 4.3.3”\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Warning message:\n",
      "“package ‘Biobase’ was built under R version 4.3.3”\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘EnhancedVolcano’ was built under R version 4.3.2”\n",
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: ggrepel\n",
      "\n",
      "Warning message:\n",
      "“package ‘clusterProfiler’ was built under R version 4.3.2”\n",
      "\n",
      "\n",
      "clusterProfiler v4.10.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141\n",
      "\n",
      "\n",
      "Attaching package: ‘clusterProfiler’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘enrichplot’ was built under R version 4.3.2”\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Warning message:\n",
      "“package ‘AnnotationDbi’ was built under R version 4.3.2”\n",
      "\n",
      "Attaching package: ‘AnnotationDbi’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘DOSE’ was built under R version 4.3.2”\n",
      "DOSE v3.28.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(DESeq2)\n",
    "library(EnhancedVolcano)\n",
    "library(clusterProfiler)\n",
    "library(enrichplot)\n",
    "library(ggplot2)\n",
    "organism = \"org.Mm.eg.db\"\n",
    "library(organism, character.only = TRUE)\n",
    "library(DOSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "df <- read.table(\"dea/sum.tsv\", header=TRUE, sep=\"\\t\")\n",
    "# Drop duplicates in id column\n",
    "df <- df[!duplicated(df$id),]\n",
    "rownames(df) <- df$id\n",
    "df$id <- NULL\n",
    "\n",
    "annotation <- read.table(\"dea/annotation.bed\", header=FALSE, sep=\"\\t\", col.names = c(\"chr\", \"start\", \"end\", \"name\", \"score\", \"strand\", \"type\", \"gene.id\", \"gene\", \"transcripts\", \"databases\"))\n",
    "# Remove chr, start, end, score, strand, gene.id, transcripts\n",
    "annotation$chr <- NULL\n",
    "annotation$start <- NULL\n",
    "annotation$end <- NULL\n",
    "annotation$score <- NULL\n",
    "annotation$strand <- NULL\n",
    "annotation$gene.id <- NULL\n",
    "annotation$transcripts <- NULL\n",
    "\n",
    "rownames(annotation) <- annotation$name\n",
    "annotation$name <- NULL\n",
    "\n",
    "phenotype <- read.csv(\"dea/phenotype.csv\", header=TRUE, row.names = 1)\n",
    "phenotype$transgene <- as.factor(phenotype$transgene)\n",
    "phenotype$drug <- as.factor(phenotype$drug)\n",
    "\n",
    "# Center and scale age and induction\n",
    "phenotype$age <- scale(phenotype$age)\n",
    "phenotype$induction <- scale(phenotype$induction)\n",
    "\n",
    "df_genes <- read.table(\"dea/gene_tpm.tsv\", sep=\"\\t\", header=TRUE, row.names=1)\n",
    "df_genes$gene_name <- NULL\n",
    "# Order genes columns like phenotype rows\n",
    "df_genes <- df_genes[,rownames(phenotype)]\n",
    "\n",
    "phenotype$esr1 <- scale(as.numeric(df_genes[\"Esr1\",]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "run_analysis <- function(directory, phenotype = NULL, design = NULL, dds = NULL, contrast = NULL, name = NULL) {\n",
    "    # Fail if both contrast and name are NULL\n",
    "    if (is.null(contrast) && is.null(name)) {\n",
    "        stop(\"Both contrast and name cannot be NULL\")\n",
    "    }\n",
    "    # Fail if none of contrast and name are NULL\n",
    "    if (!is.null(contrast) && !is.null(name)) {\n",
    "        stop(\"Both contrast and name cannot be provided\")\n",
    "    }\n",
    "\n",
    "    if (is.null(phenotype) && is.null(design) && is.null(dds)) {\n",
    "        stop(\"Either phenotype and design or dds must be provided\")\n",
    "    }\n",
    "\n",
    "    if (is.null(phenotype) != is.null(design)) {\n",
    "        stop(\"Both phenotype and design must be provided\")\n",
    "    }\n",
    "\n",
    "    if (!is.null(dds) && !is.null(phenotype)) {\n",
    "        stop(\"Either dds or phenotype and design must be provided\")\n",
    "    }\n",
    "\n",
    "    alpha <- 0.05\n",
    "\n",
    "    if (is.null(dds)) {\n",
    "        dds <- DESeqDataSetFromMatrix(countData = df[, rownames(phenotype)],\n",
    "                                    colData = phenotype,\n",
    "                                    design = design)\n",
    "        dds <- DESeq(dds)\n",
    "    }\n",
    "\n",
    "    if (!is.null(contrast)) {\n",
    "        res <- results(dds, contrast = contrast, alpha = alpha, lfcThreshold = 2, altHypothesis = \"greaterAbs\")\n",
    "    } else {\n",
    "        res <- results(dds, name = name, alpha = alpha, lfcThreshold = 2, altHypothesis = \"greaterAbs\")\n",
    "    }\n",
    "    res <- res[order(res$padj),]\n",
    "    res <- cbind(res, annotation[rownames(res),])\n",
    "\n",
    "    # Recreate directory if it exists\n",
    "    if (dir.exists(directory)) {\n",
    "        unlink(directory, recursive = TRUE)\n",
    "    }\n",
    "    dir.create(directory, showWarnings = FALSE)\n",
    "\n",
    "    write.table(res, file=paste0(directory, \"/res.tsv\"), sep=\"\\t\", col.names=NA, row.names=TRUE, quote=FALSE)\n",
    "\n",
    "    EnhancedVolcano(res,\n",
    "        x='log2FoldChange',\n",
    "        y='padj',\n",
    "        lab=res$gene,\n",
    "        pCutoff = alpha)\n",
    "    ggsave(paste0(directory, \"/volcano.png\"))\n",
    "\n",
    "    # Keep only rows without any NA\n",
    "    res <- res[complete.cases(res),]\n",
    "\n",
    "    sign <- res[res$padj < alpha,]\n",
    "\n",
    "    genes <- unique(unlist(strsplit(sign$gene, \",\")))\n",
    "    write.table(genes, file=paste0(directory, \"/genes.txt\"), row.names=FALSE, col.names=FALSE, quote=FALSE)\n",
    "    db <- unique(unlist(strsplit(sign$databases, \",\")))\n",
    "    write.table(db, file=paste0(directory, \"/db.txt\"), row.names=FALSE, col.names=FALSE, quote=FALSE)\n",
    "\n",
    "    gene_list <- sign$log2FoldChange\n",
    "    names(gene_list) <- sign$gene\n",
    "    gene_list = sort(gene_list, decreasing = TRUE)\n",
    "\n",
    "    gse <- gseGO(geneList=gene_list,\n",
    "            ont =\"ALL\", \n",
    "            keyType = \"SYMBOL\",\n",
    "            pvalueCutoff = 0.05,\n",
    "            verbose = TRUE,\n",
    "            OrgDb = organism,\n",
    "            pAdjustMethod = \"BH\")\n",
    "\n",
    "    # If no significant GO terms, return\n",
    "    if (nrow(as.data.frame(gse)) > 0) {\n",
    "        dotplot(gse, showCategory=10, split=\".sign\") + facet_grid(.~.sign)\n",
    "        ggsave(paste0(directory, \"/dot.png\"))\n",
    "\n",
    "        gse <- pairwise_termsim(gse)\n",
    "        emapplot(gse, showCategory = 10)\n",
    "        ggsave(paste0(directory, \"/emap.png\"))\n",
    "\n",
    "        cnetplot(gse, categorySize=\"pvalue\", foldChange=gene_list, showCategory = 3)\n",
    "        ggsave(paste0(directory, \"/cnet.png\"))\n",
    "\n",
    "        ridgeplot(gse) + labs(x = \"enrichment distribution\")\n",
    "        ggsave(paste0(directory, \"/ridge.png\"))\n",
    "    } else {\n",
    "        print(\"No significant GO terms\")\n",
    "    }\n",
    "\n",
    "    return(dds)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Aging"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "5 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (41.78% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"No significant GO terms\"\n"
     ]
    }
   ],
   "source": [
    "dds_esr1 <- run_analysis(\n",
    "    \"dea/age\",\n",
    "    phenotype = phenotype,\n",
    "    design = ~ age + transgene + induction + drug + esr1,\n",
    "    name = \"age\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "0.0136188807920855"
      ],
      "text/latex": [
       "0.0136188807920855"
      ],
      "text/markdown": [
       "0.0136188807920855"
      ],
      "text/plain": [
       "[1] 0.01361888"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cor.test(as.numeric(df[1,]), phenotype$esr1)$p.value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ESR1 association"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 34512 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>correlation</th><th scope=col>p.value</th><th scope=col>p.adj</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>chr14:87047771-87072537:-</th><td>-0.3978165</td><td>0.0005391544</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr15:44259906-44269388:-</th><td>-0.3899490</td><td>0.0007091335</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr3:55588318-55631070:-</th><td>-0.3779405</td><td>0.0010637002</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr7:19266598-19266682:+</th><td>-0.3616702</td><td>0.0017989426</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr7:101794764-101825612:-</th><td>-0.3536681</td><td>0.0023067755</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr8:121472376-121474017:-</th><td>-0.3443321</td><td>0.0030588371</td><td>0.3051057</td></tr>\n",
       "\t<tr><th scope=row>chr9:118455515-118465694:+</th><td>-0.3429474</td><td>0.0031872924</td><td>0.3151858</td></tr>\n",
       "\t<tr><th scope=row>chr10:39680947-39690086:+</th><td>-0.3415289</td><td>0.0033238563</td><td>0.3277512</td></tr>\n",
       "\t<tr><th scope=row>chr9:75202443-75203442:+</th><td>-0.3393938</td><td>0.0035392431</td><td>0.3460237</td></tr>\n",
       "\t<tr><th scope=row>chr8:106822795-106830783:+</th><td>-0.3390681</td><td>0.0035731721</td><td>0.3463970</td></tr>\n",
       "\t<tr><th scope=row>chr10:61496691-61501245:+</th><td>-0.3384812</td><td>0.0036350368</td><td>0.3493103</td></tr>\n",
       "\t<tr><th scope=row>chr9:7023310-7049192:-</th><td>-0.3344770</td><td>0.0040831479</td><td>0.3808584</td></tr>\n",
       "\t<tr><th scope=row>chr11:119343336-119358906:+</th><td>-0.3339527</td><td>0.0041453152</td><td>0.3845783</td></tr>\n",
       "\t<tr><th scope=row>chr14:47786376-47792098:-</th><td>-0.3313751</td><td>0.0044633250</td><td>0.4085896</td></tr>\n",
       "\t<tr><th scope=row>chr16:11017248-11017583:-</th><td>-0.3313751</td><td>0.0044633250</td><td>0.4085896</td></tr>\n",
       "\t<tr><th scope=row>chr2:153232074-153232159:-</th><td>-0.3305739</td><td>0.0045664825</td><td>0.4169271</td></tr>\n",
       "\t<tr><th scope=row>chr1:134403526-134413651:+</th><td>-0.3253252</td><td>0.0052960942</td><td>0.4703549</td></tr>\n",
       "\t<tr><th scope=row>chr12:33185965-33191115:-</th><td>-0.3225245</td><td>0.0057260889</td><td>0.5008217</td></tr>\n",
       "\t<tr><th scope=row>chr4:132912280-132913063:+</th><td>-0.3225121</td><td>0.0057280628</td><td>0.5008217</td></tr>\n",
       "\t<tr><th scope=row>chr2:37470685-37517538:-</th><td>-0.3200430</td><td>0.0061324989</td><td>0.5304381</td></tr>\n",
       "\t<tr><th scope=row>chr5:43562277-43562424:+</th><td>-0.3164231</td><td>0.0067709345</td><td>0.5784121</td></tr>\n",
       "\t<tr><th scope=row>chr6:70703770-70703879:-</th><td>-0.3156952</td><td>0.0069061446</td><td>0.5813289</td></tr>\n",
       "\t<tr><th scope=row>chr15:79304104-79305210:-</th><td>-0.3130931</td><td>0.0074090827</td><td>0.6125360</td></tr>\n",
       "\t<tr><th scope=row>chr4:144633587-144633673:-</th><td>-0.3130440</td><td>0.0074188705</td><td>0.6125360</td></tr>\n",
       "\t<tr><th scope=row>chr4:6822905-6842426:-</th><td>-0.3119398</td><td>0.0076420961</td><td>0.6279620</td></tr>\n",
       "\t<tr><th scope=row>chr9:35146526-35156002:-</th><td>-0.3109719</td><td>0.0078425840</td><td>0.6323908</td></tr>\n",
       "\t<tr><th scope=row>chr9:22555040-22582253:+</th><td>-0.3093990</td><td>0.0081782111</td><td>0.6548641</td></tr>\n",
       "\t<tr><th scope=row>chr16:35752833-35752940:+</th><td>-0.3091089</td><td>0.0082414883</td><td>0.6573943</td></tr>\n",
       "\t<tr><th scope=row>chr16:84821741-84822385:-</th><td>-0.3073011</td><td>0.0086454374</td><td>0.6573943</td></tr>\n",
       "\t<tr><th scope=row>chr6:149084892-149086622:-</th><td>-0.3067457</td><td>0.0087729557</td><td>0.6573943</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>chr6:86424293-86442945:+</th><td>0.3502473</td><td>0.0025605591</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr15:57915620-57927576:-</th><td>0.3506429</td><td>0.0025299879</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr4:44019874-44025560:+</th><td>0.3539196</td><td>0.0022890394</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr1:190541079-190631768:-</th><td>0.3570192</td><td>0.0020802589</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr16:31962954-31963163:+</th><td>0.3570192</td><td>0.0020802589</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr15:4064741-4066663:+</th><td>0.3582595</td><td>0.0020016176</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr7:100922140-100922233:-</th><td>0.3591942</td><td>0.0019441320</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr5:34000456-34003611:+</th><td>0.3601556</td><td>0.0018865509</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr7:143077898-143081048:-</th><td>0.3602110</td><td>0.0018832794</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr12:82404006-82419316:+</th><td>0.3629426</td><td>0.0017281918</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr12:91460078-91486119:+</th><td>0.3637509</td><td>0.0016845599</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr3:51207203-51233453:-</th><td>0.3655671</td><td>0.0015901101</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr9:55593163-55767202:-</th><td>0.3675732</td><td>0.0014913495</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr3:68993019-69002948:-</th><td>0.3680199</td><td>0.0014701273</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr5:3819429-3822787:-</th><td>0.3680199</td><td>0.0014701273</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr11:30099506-30099608:+</th><td>0.3686532</td><td>0.0014405041</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr1:176906340-177076447:-</th><td>0.3688228</td><td>0.0014326627</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr11:50198296-50199756:-</th><td>0.3742237</td><td>0.0012022300</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr8:3267121-3267200:+</th><td>0.3759669</td><td>0.0011353481</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr8:4315516-4315646:+</th><td>0.3782896</td><td>0.0010514610</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr11:62008525-62010108:+</th><td>0.3852407</td><td>0.0008328420</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr3:88873119-88893097:+</th><td>0.3859437</td><td>0.0008132063</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr2:91047011-91049234:-</th><td>0.3880925</td><td>0.0007557648</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr18:61290415-61290833:-</th><td>0.3905112</td><td>0.0006955370</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr5:34332177-34332279:+</th><td>0.3912835</td><td>0.0006772459</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr2:91596791-91604092:+</th><td>0.3932486</td><td>0.0006326593</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr9:9046505-9059160:-</th><td>0.3965100</td><td>0.0005645226</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr5:34332175-34332280:+</th><td>0.4083013</td><td>0.0003702455</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr12:98724770-98726669:+</th><td>0.4174379</td><td>0.0002641440</td><td>0.2632107</td></tr>\n",
       "\t<tr><th scope=row>chr14:120562206-120562824:+</th><td>0.4187837</td><td>0.0002511225</td><td>0.2632107</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 34512 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & correlation & p.value & p.adj\\\\\n",
       "  & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tchr14:87047771-87072537:- & -0.3978165 & 0.0005391544 & 0.2632107\\\\\n",
       "\tchr15:44259906-44269388:- & -0.3899490 & 0.0007091335 & 0.2632107\\\\\n",
       "\tchr3:55588318-55631070:- & -0.3779405 & 0.0010637002 & 0.2632107\\\\\n",
       "\tchr7:19266598-19266682:+ & -0.3616702 & 0.0017989426 & 0.2632107\\\\\n",
       "\tchr7:101794764-101825612:- & -0.3536681 & 0.0023067755 & 0.2632107\\\\\n",
       "\tchr8:121472376-121474017:- & -0.3443321 & 0.0030588371 & 0.3051057\\\\\n",
       "\tchr9:118455515-118465694:+ & -0.3429474 & 0.0031872924 & 0.3151858\\\\\n",
       "\tchr10:39680947-39690086:+ & -0.3415289 & 0.0033238563 & 0.3277512\\\\\n",
       "\tchr9:75202443-75203442:+ & -0.3393938 & 0.0035392431 & 0.3460237\\\\\n",
       "\tchr8:106822795-106830783:+ & -0.3390681 & 0.0035731721 & 0.3463970\\\\\n",
       "\tchr10:61496691-61501245:+ & -0.3384812 & 0.0036350368 & 0.3493103\\\\\n",
       "\tchr9:7023310-7049192:- & -0.3344770 & 0.0040831479 & 0.3808584\\\\\n",
       "\tchr11:119343336-119358906:+ & -0.3339527 & 0.0041453152 & 0.3845783\\\\\n",
       "\tchr14:47786376-47792098:- & -0.3313751 & 0.0044633250 & 0.4085896\\\\\n",
       "\tchr16:11017248-11017583:- & -0.3313751 & 0.0044633250 & 0.4085896\\\\\n",
       "\tchr2:153232074-153232159:- & -0.3305739 & 0.0045664825 & 0.4169271\\\\\n",
       "\tchr1:134403526-134413651:+ & -0.3253252 & 0.0052960942 & 0.4703549\\\\\n",
       "\tchr12:33185965-33191115:- & -0.3225245 & 0.0057260889 & 0.5008217\\\\\n",
       "\tchr4:132912280-132913063:+ & -0.3225121 & 0.0057280628 & 0.5008217\\\\\n",
       "\tchr2:37470685-37517538:- & -0.3200430 & 0.0061324989 & 0.5304381\\\\\n",
       "\tchr5:43562277-43562424:+ & -0.3164231 & 0.0067709345 & 0.5784121\\\\\n",
       "\tchr6:70703770-70703879:- & -0.3156952 & 0.0069061446 & 0.5813289\\\\\n",
       "\tchr15:79304104-79305210:- & -0.3130931 & 0.0074090827 & 0.6125360\\\\\n",
       "\tchr4:144633587-144633673:- & -0.3130440 & 0.0074188705 & 0.6125360\\\\\n",
       "\tchr4:6822905-6842426:- & -0.3119398 & 0.0076420961 & 0.6279620\\\\\n",
       "\tchr9:35146526-35156002:- & -0.3109719 & 0.0078425840 & 0.6323908\\\\\n",
       "\tchr9:22555040-22582253:+ & -0.3093990 & 0.0081782111 & 0.6548641\\\\\n",
       "\tchr16:35752833-35752940:+ & -0.3091089 & 0.0082414883 & 0.6573943\\\\\n",
       "\tchr16:84821741-84822385:- & -0.3073011 & 0.0086454374 & 0.6573943\\\\\n",
       "\tchr6:149084892-149086622:- & -0.3067457 & 0.0087729557 & 0.6573943\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tchr6:86424293-86442945:+ & 0.3502473 & 0.0025605591 & 0.2632107\\\\\n",
       "\tchr15:57915620-57927576:- & 0.3506429 & 0.0025299879 & 0.2632107\\\\\n",
       "\tchr4:44019874-44025560:+ & 0.3539196 & 0.0022890394 & 0.2632107\\\\\n",
       "\tchr1:190541079-190631768:- & 0.3570192 & 0.0020802589 & 0.2632107\\\\\n",
       "\tchr16:31962954-31963163:+ & 0.3570192 & 0.0020802589 & 0.2632107\\\\\n",
       "\tchr15:4064741-4066663:+ & 0.3582595 & 0.0020016176 & 0.2632107\\\\\n",
       "\tchr7:100922140-100922233:- & 0.3591942 & 0.0019441320 & 0.2632107\\\\\n",
       "\tchr5:34000456-34003611:+ & 0.3601556 & 0.0018865509 & 0.2632107\\\\\n",
       "\tchr7:143077898-143081048:- & 0.3602110 & 0.0018832794 & 0.2632107\\\\\n",
       "\tchr12:82404006-82419316:+ & 0.3629426 & 0.0017281918 & 0.2632107\\\\\n",
       "\tchr12:91460078-91486119:+ & 0.3637509 & 0.0016845599 & 0.2632107\\\\\n",
       "\tchr3:51207203-51233453:- & 0.3655671 & 0.0015901101 & 0.2632107\\\\\n",
       "\tchr9:55593163-55767202:- & 0.3675732 & 0.0014913495 & 0.2632107\\\\\n",
       "\tchr3:68993019-69002948:- & 0.3680199 & 0.0014701273 & 0.2632107\\\\\n",
       "\tchr5:3819429-3822787:- & 0.3680199 & 0.0014701273 & 0.2632107\\\\\n",
       "\tchr11:30099506-30099608:+ & 0.3686532 & 0.0014405041 & 0.2632107\\\\\n",
       "\tchr1:176906340-177076447:- & 0.3688228 & 0.0014326627 & 0.2632107\\\\\n",
       "\tchr11:50198296-50199756:- & 0.3742237 & 0.0012022300 & 0.2632107\\\\\n",
       "\tchr8:3267121-3267200:+ & 0.3759669 & 0.0011353481 & 0.2632107\\\\\n",
       "\tchr8:4315516-4315646:+ & 0.3782896 & 0.0010514610 & 0.2632107\\\\\n",
       "\tchr11:62008525-62010108:+ & 0.3852407 & 0.0008328420 & 0.2632107\\\\\n",
       "\tchr3:88873119-88893097:+ & 0.3859437 & 0.0008132063 & 0.2632107\\\\\n",
       "\tchr2:91047011-91049234:- & 0.3880925 & 0.0007557648 & 0.2632107\\\\\n",
       "\tchr18:61290415-61290833:- & 0.3905112 & 0.0006955370 & 0.2632107\\\\\n",
       "\tchr5:34332177-34332279:+ & 0.3912835 & 0.0006772459 & 0.2632107\\\\\n",
       "\tchr2:91596791-91604092:+ & 0.3932486 & 0.0006326593 & 0.2632107\\\\\n",
       "\tchr9:9046505-9059160:- & 0.3965100 & 0.0005645226 & 0.2632107\\\\\n",
       "\tchr5:34332175-34332280:+ & 0.4083013 & 0.0003702455 & 0.2632107\\\\\n",
       "\tchr12:98724770-98726669:+ & 0.4174379 & 0.0002641440 & 0.2632107\\\\\n",
       "\tchr14:120562206-120562824:+ & 0.4187837 & 0.0002511225 & 0.2632107\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 34512 × 3\n",
       "\n",
       "| <!--/--> | correlation &lt;dbl&gt; | p.value &lt;dbl&gt; | p.adj &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| chr14:87047771-87072537:- | -0.3978165 | 0.0005391544 | 0.2632107 |\n",
       "| chr15:44259906-44269388:- | -0.3899490 | 0.0007091335 | 0.2632107 |\n",
       "| chr3:55588318-55631070:- | -0.3779405 | 0.0010637002 | 0.2632107 |\n",
       "| chr7:19266598-19266682:+ | -0.3616702 | 0.0017989426 | 0.2632107 |\n",
       "| chr7:101794764-101825612:- | -0.3536681 | 0.0023067755 | 0.2632107 |\n",
       "| chr8:121472376-121474017:- | -0.3443321 | 0.0030588371 | 0.3051057 |\n",
       "| chr9:118455515-118465694:+ | -0.3429474 | 0.0031872924 | 0.3151858 |\n",
       "| chr10:39680947-39690086:+ | -0.3415289 | 0.0033238563 | 0.3277512 |\n",
       "| chr9:75202443-75203442:+ | -0.3393938 | 0.0035392431 | 0.3460237 |\n",
       "| chr8:106822795-106830783:+ | -0.3390681 | 0.0035731721 | 0.3463970 |\n",
       "| chr10:61496691-61501245:+ | -0.3384812 | 0.0036350368 | 0.3493103 |\n",
       "| chr9:7023310-7049192:- | -0.3344770 | 0.0040831479 | 0.3808584 |\n",
       "| chr11:119343336-119358906:+ | -0.3339527 | 0.0041453152 | 0.3845783 |\n",
       "| chr14:47786376-47792098:- | -0.3313751 | 0.0044633250 | 0.4085896 |\n",
       "| chr16:11017248-11017583:- | -0.3313751 | 0.0044633250 | 0.4085896 |\n",
       "| chr2:153232074-153232159:- | -0.3305739 | 0.0045664825 | 0.4169271 |\n",
       "| chr1:134403526-134413651:+ | -0.3253252 | 0.0052960942 | 0.4703549 |\n",
       "| chr12:33185965-33191115:- | -0.3225245 | 0.0057260889 | 0.5008217 |\n",
       "| chr4:132912280-132913063:+ | -0.3225121 | 0.0057280628 | 0.5008217 |\n",
       "| chr2:37470685-37517538:- | -0.3200430 | 0.0061324989 | 0.5304381 |\n",
       "| chr5:43562277-43562424:+ | -0.3164231 | 0.0067709345 | 0.5784121 |\n",
       "| chr6:70703770-70703879:- | -0.3156952 | 0.0069061446 | 0.5813289 |\n",
       "| chr15:79304104-79305210:- | -0.3130931 | 0.0074090827 | 0.6125360 |\n",
       "| chr4:144633587-144633673:- | -0.3130440 | 0.0074188705 | 0.6125360 |\n",
       "| chr4:6822905-6842426:- | -0.3119398 | 0.0076420961 | 0.6279620 |\n",
       "| chr9:35146526-35156002:- | -0.3109719 | 0.0078425840 | 0.6323908 |\n",
       "| chr9:22555040-22582253:+ | -0.3093990 | 0.0081782111 | 0.6548641 |\n",
       "| chr16:35752833-35752940:+ | -0.3091089 | 0.0082414883 | 0.6573943 |\n",
       "| chr16:84821741-84822385:- | -0.3073011 | 0.0086454374 | 0.6573943 |\n",
       "| chr6:149084892-149086622:- | -0.3067457 | 0.0087729557 | 0.6573943 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| chr6:86424293-86442945:+ | 0.3502473 | 0.0025605591 | 0.2632107 |\n",
       "| chr15:57915620-57927576:- | 0.3506429 | 0.0025299879 | 0.2632107 |\n",
       "| chr4:44019874-44025560:+ | 0.3539196 | 0.0022890394 | 0.2632107 |\n",
       "| chr1:190541079-190631768:- | 0.3570192 | 0.0020802589 | 0.2632107 |\n",
       "| chr16:31962954-31963163:+ | 0.3570192 | 0.0020802589 | 0.2632107 |\n",
       "| chr15:4064741-4066663:+ | 0.3582595 | 0.0020016176 | 0.2632107 |\n",
       "| chr7:100922140-100922233:- | 0.3591942 | 0.0019441320 | 0.2632107 |\n",
       "| chr5:34000456-34003611:+ | 0.3601556 | 0.0018865509 | 0.2632107 |\n",
       "| chr7:143077898-143081048:- | 0.3602110 | 0.0018832794 | 0.2632107 |\n",
       "| chr12:82404006-82419316:+ | 0.3629426 | 0.0017281918 | 0.2632107 |\n",
       "| chr12:91460078-91486119:+ | 0.3637509 | 0.0016845599 | 0.2632107 |\n",
       "| chr3:51207203-51233453:- | 0.3655671 | 0.0015901101 | 0.2632107 |\n",
       "| chr9:55593163-55767202:- | 0.3675732 | 0.0014913495 | 0.2632107 |\n",
       "| chr3:68993019-69002948:- | 0.3680199 | 0.0014701273 | 0.2632107 |\n",
       "| chr5:3819429-3822787:- | 0.3680199 | 0.0014701273 | 0.2632107 |\n",
       "| chr11:30099506-30099608:+ | 0.3686532 | 0.0014405041 | 0.2632107 |\n",
       "| chr1:176906340-177076447:- | 0.3688228 | 0.0014326627 | 0.2632107 |\n",
       "| chr11:50198296-50199756:- | 0.3742237 | 0.0012022300 | 0.2632107 |\n",
       "| chr8:3267121-3267200:+ | 0.3759669 | 0.0011353481 | 0.2632107 |\n",
       "| chr8:4315516-4315646:+ | 0.3782896 | 0.0010514610 | 0.2632107 |\n",
       "| chr11:62008525-62010108:+ | 0.3852407 | 0.0008328420 | 0.2632107 |\n",
       "| chr3:88873119-88893097:+ | 0.3859437 | 0.0008132063 | 0.2632107 |\n",
       "| chr2:91047011-91049234:- | 0.3880925 | 0.0007557648 | 0.2632107 |\n",
       "| chr18:61290415-61290833:- | 0.3905112 | 0.0006955370 | 0.2632107 |\n",
       "| chr5:34332177-34332279:+ | 0.3912835 | 0.0006772459 | 0.2632107 |\n",
       "| chr2:91596791-91604092:+ | 0.3932486 | 0.0006326593 | 0.2632107 |\n",
       "| chr9:9046505-9059160:- | 0.3965100 | 0.0005645226 | 0.2632107 |\n",
       "| chr5:34332175-34332280:+ | 0.4083013 | 0.0003702455 | 0.2632107 |\n",
       "| chr12:98724770-98726669:+ | 0.4174379 | 0.0002641440 | 0.2632107 |\n",
       "| chr14:120562206-120562824:+ | 0.4187837 | 0.0002511225 | 0.2632107 |\n",
       "\n"
      ],
      "text/plain": [
       "                            correlation p.value      p.adj    \n",
       "chr14:87047771-87072537:-   -0.3978165  0.0005391544 0.2632107\n",
       "chr15:44259906-44269388:-   -0.3899490  0.0007091335 0.2632107\n",
       "chr3:55588318-55631070:-    -0.3779405  0.0010637002 0.2632107\n",
       "chr7:19266598-19266682:+    -0.3616702  0.0017989426 0.2632107\n",
       "chr7:101794764-101825612:-  -0.3536681  0.0023067755 0.2632107\n",
       "chr8:121472376-121474017:-  -0.3443321  0.0030588371 0.3051057\n",
       "chr9:118455515-118465694:+  -0.3429474  0.0031872924 0.3151858\n",
       "chr10:39680947-39690086:+   -0.3415289  0.0033238563 0.3277512\n",
       "chr9:75202443-75203442:+    -0.3393938  0.0035392431 0.3460237\n",
       "chr8:106822795-106830783:+  -0.3390681  0.0035731721 0.3463970\n",
       "chr10:61496691-61501245:+   -0.3384812  0.0036350368 0.3493103\n",
       "chr9:7023310-7049192:-      -0.3344770  0.0040831479 0.3808584\n",
       "chr11:119343336-119358906:+ -0.3339527  0.0041453152 0.3845783\n",
       "chr14:47786376-47792098:-   -0.3313751  0.0044633250 0.4085896\n",
       "chr16:11017248-11017583:-   -0.3313751  0.0044633250 0.4085896\n",
       "chr2:153232074-153232159:-  -0.3305739  0.0045664825 0.4169271\n",
       "chr1:134403526-134413651:+  -0.3253252  0.0052960942 0.4703549\n",
       "chr12:33185965-33191115:-   -0.3225245  0.0057260889 0.5008217\n",
       "chr4:132912280-132913063:+  -0.3225121  0.0057280628 0.5008217\n",
       "chr2:37470685-37517538:-    -0.3200430  0.0061324989 0.5304381\n",
       "chr5:43562277-43562424:+    -0.3164231  0.0067709345 0.5784121\n",
       "chr6:70703770-70703879:-    -0.3156952  0.0069061446 0.5813289\n",
       "chr15:79304104-79305210:-   -0.3130931  0.0074090827 0.6125360\n",
       "chr4:144633587-144633673:-  -0.3130440  0.0074188705 0.6125360\n",
       "chr4:6822905-6842426:-      -0.3119398  0.0076420961 0.6279620\n",
       "chr9:35146526-35156002:-    -0.3109719  0.0078425840 0.6323908\n",
       "chr9:22555040-22582253:+    -0.3093990  0.0081782111 0.6548641\n",
       "chr16:35752833-35752940:+   -0.3091089  0.0082414883 0.6573943\n",
       "chr16:84821741-84822385:-   -0.3073011  0.0086454374 0.6573943\n",
       "chr6:149084892-149086622:-  -0.3067457  0.0087729557 0.6573943\n",
       "⋮                           ⋮           ⋮            ⋮        \n",
       "chr6:86424293-86442945:+    0.3502473   0.0025605591 0.2632107\n",
       "chr15:57915620-57927576:-   0.3506429   0.0025299879 0.2632107\n",
       "chr4:44019874-44025560:+    0.3539196   0.0022890394 0.2632107\n",
       "chr1:190541079-190631768:-  0.3570192   0.0020802589 0.2632107\n",
       "chr16:31962954-31963163:+   0.3570192   0.0020802589 0.2632107\n",
       "chr15:4064741-4066663:+     0.3582595   0.0020016176 0.2632107\n",
       "chr7:100922140-100922233:-  0.3591942   0.0019441320 0.2632107\n",
       "chr5:34000456-34003611:+    0.3601556   0.0018865509 0.2632107\n",
       "chr7:143077898-143081048:-  0.3602110   0.0018832794 0.2632107\n",
       "chr12:82404006-82419316:+   0.3629426   0.0017281918 0.2632107\n",
       "chr12:91460078-91486119:+   0.3637509   0.0016845599 0.2632107\n",
       "chr3:51207203-51233453:-    0.3655671   0.0015901101 0.2632107\n",
       "chr9:55593163-55767202:-    0.3675732   0.0014913495 0.2632107\n",
       "chr3:68993019-69002948:-    0.3680199   0.0014701273 0.2632107\n",
       "chr5:3819429-3822787:-      0.3680199   0.0014701273 0.2632107\n",
       "chr11:30099506-30099608:+   0.3686532   0.0014405041 0.2632107\n",
       "chr1:176906340-177076447:-  0.3688228   0.0014326627 0.2632107\n",
       "chr11:50198296-50199756:-   0.3742237   0.0012022300 0.2632107\n",
       "chr8:3267121-3267200:+      0.3759669   0.0011353481 0.2632107\n",
       "chr8:4315516-4315646:+      0.3782896   0.0010514610 0.2632107\n",
       "chr11:62008525-62010108:+   0.3852407   0.0008328420 0.2632107\n",
       "chr3:88873119-88893097:+    0.3859437   0.0008132063 0.2632107\n",
       "chr2:91047011-91049234:-    0.3880925   0.0007557648 0.2632107\n",
       "chr18:61290415-61290833:-   0.3905112   0.0006955370 0.2632107\n",
       "chr5:34332177-34332279:+    0.3912835   0.0006772459 0.2632107\n",
       "chr2:91596791-91604092:+    0.3932486   0.0006326593 0.2632107\n",
       "chr9:9046505-9059160:-      0.3965100   0.0005645226 0.2632107\n",
       "chr5:34332175-34332280:+    0.4083013   0.0003702455 0.2632107\n",
       "chr12:98724770-98726669:+   0.4174379   0.0002641440 0.2632107\n",
       "chr14:120562206-120562824:+ 0.4187837   0.0002511225 0.2632107"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "correlations <- t(apply(df, 1, function(row) {\n",
    "    res <- cor.test(as.numeric(row), phenotype$esr1)\n",
    "    return(c(res$estimate, res$p.value))\n",
    "}))\n",
    "correlations <- as.data.frame(correlations)\n",
    "colnames(correlations) <- c(\"correlation\", \"p.value\")\n",
    "correlations$p.adj <- p.adjust(correlations$p.value, method = \"BH\")\n",
    "correlations <- correlations[order(correlations$correlation),]\n",
    "correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (65.45% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"No significant GO terms\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 34512 72 \n",
       "metadata(1): version\n",
       "assays(4): counts mu H cooks\n",
       "rownames(34512): chr1:3451731-3451944:+ chr1:3638591-3638734:+ ...\n",
       "  chrX_GL456233v2_random:409324-409604:+\n",
       "  chrX_GL456233v2_random:409324-422124:+\n",
       "rowData names(42): baseMean baseVar ... deviance maxCooks\n",
       "colnames(72): aging_12m_ESR1_no_1 aging_12m_ESR1_no_2 ...\n",
       "  antiHormonal_18m_ESR1_no_2 antiHormonal_18m_ESR1_no_3\n",
       "colData names(8): condition age ... esr1 sizeFactor"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "run_analysis(\n",
    "    \"dea/esr1\",\n",
    "    dds = dds_esr1,\n",
    "    name = \"esr1\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Drug effects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Tamoxifen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (3.12% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Warning message in cnetplot.enrichResult(x, ...):\n",
      "“Use 'color.params = list(foldChange = your_value)' instead of 'foldChange'.\n",
      " The foldChange parameter will be removed in the next version.”\n",
      "\u001b[1m\u001b[22mScale for \u001b[32msize\u001b[39m is already present.\n",
      "Adding another scale for \u001b[32msize\u001b[39m, which will replace the existing scale.\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "Picking joint bandwidth of 1.27\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dds_no_esr1 <- run_analysis(\n",
    "    \"dea/tamoxifen\",\n",
    "    phenotype = phenotype,\n",
    "    design = ~ age + transgene + induction + drug,\n",
    "    contrast=c(\"drug\", \"tamoxifen\", \"no\")\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Letrozole"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mSaving 6.67 x 6.67 in image\n",
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are duplicate gene names, fgsea may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"No significant GO terms\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 34512 72 \n",
       "metadata(1): version\n",
       "assays(4): counts mu H cooks\n",
       "rownames(34512): chr1:3451731-3451944:+ chr1:3638591-3638734:+ ...\n",
       "  chrX_GL456233v2_random:409324-409604:+\n",
       "  chrX_GL456233v2_random:409324-422124:+\n",
       "rowData names(38): baseMean baseVar ... deviance maxCooks\n",
       "colnames(72): aging_12m_ESR1_no_1 aging_12m_ESR1_no_2 ...\n",
       "  antiHormonal_18m_ESR1_no_2 antiHormonal_18m_ESR1_no_3\n",
       "colData names(8): condition age ... esr1 sizeFactor"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "run_analysis(\n",
    "    \"dea/letrozole\",\n",
    "    dds = dds_no_esr1,\n",
    "    contrast=c(\"drug\", \"letrozole\", \"no\")\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
