\chapter{Results and Discussion}

From a molecular perspective, circRNAs are similar to their linear
counterparts, with the primary distinction being their circular structure.
As explained in \cref{sec:circrna_biogenesis}, circRNAs are derived from a
linear precursor (identical to linear RNA) that undergoes back-splicing to form
a loop.
Thus, the only way to computationally identify circRNAs is by detecting reads
spanning the back-splice junction (BSJ), which cannot be accounted for by
canonical forward splicing.
Several tools for BSJ detection are discussed in
\cref{subsec:circrna_detection}.

However, detecting BSJs only pinpoints the start and end of a circRNA within
the genome, without revealing its internal structure.
To reconstruct the full circRNA sequence, a de novo assembly of the circular
transcriptome that accounts for back-splice junctions is required.
Tools like CIRI-full, circRNAfull, and JCcirc are capable of this task, but
they rely on paired-end or long-read sequencing data.
To date, no tools are available for de novo assembly of circRNAs using
single-end short-read data.
Since the data used in this thesis is single-end, the focus is limited to the
detection, quantification, and analysis of BSJs.

\include{chapters/4_results_and_discussion/1_bsj_detection.tex}
\include{chapters/4_results_and_discussion/2_quantification.tex}
\include{chapters/4_results_and_discussion/3_de_analysis.tex}
\include{chapters/4_results_and_discussion/4_interpretation.tex}
