\chapter{Results and Discussion}

Several tools for \gls{bsj} detection are discussed in
\cref{subsec:circrna_detection}.

However, detecting \gls{bsj}s only reveals the start and end positions of a
circRNA within the genome, without providing information on its internal
structure.
To reconstruct the full circRNA sequence, a de novo assembly of the circular
transcriptome that accounts for \gls{bsj}s is necessary.
Tools like CIRI-full, circRNAfull, JCcirc, and psirc can achieve this, but they
rely on paired-end or long-read sequencing data.
As of now, no tools are available for de novo assembly of circRNAs using
single-end short-read data, to my knowledge.
Since the data used in this thesis is single-end, the focus will be limited to
the detection, quantification, and analysis of \gls{bsj}s.

\input{chapters/4_results_and_discussion/1_bsj_detection.tex}
\input{chapters/4_results_and_discussion/2_quantification.tex}
\input{chapters/4_results_and_discussion/3_de_analysis.tex}
\input{chapters/4_results_and_discussion/4_interpretation.tex}
